crates/adapters/hyperliquid/src/data/mod.rs

```diff
@@ -145,9 +145,67 @@ impl HyperliquidDataClient {
             .await
             .context("Failed to connect to Hyperliquid WebSocket")?;
 
+        // No message processing loop needed - the WebSocket client handles it internally
         Ok(())
     }
 
+    #[allow(dead_code)]
+    fn handle_ws_message(
+        msg: HyperliquidWsMessage,
+        data_sender: &tokio::sync::mpsc::UnboundedSender<DataEvent>,
+        instruments: &Arc<RwLock<AHashMap<InstrumentId, InstrumentAny>>>,
+        _venue: Venue,
+        clock: &'static AtomicTime,
+    ) {
+        use crate::websocket::parse::parse_ws_quote_tick;
+
+        match msg {
+            HyperliquidWsMessage::Bbo { data } => {
+                tracing::debug!("Received BBO message for coin: {}", data.coin);
+
+                // Find instrument by matching the coin prefix
+                // Hyperliquid WebSocket sends coin="BTC", but we have instrument "BTC-PERP"
+                let instruments_map = instruments.read().unwrap();
+                let matching_instrument = instruments_map.iter().find(|(id, _)| {
```

Comment:
Just flagging perf for later, fairly expensive iteration on every ws message.

Reply:
Addressed here https://github.com/nautechsystems/nautilus_trader/pull/3089
Thanks